tidy3d/plugins/klayout/drc/drc.py

```diff
@@ -0,0 +1,310 @@
+        """Check GDS filetype is .gds"""
+        if v.suffix != ".gds":
+            raise ValidationError(f"GDS file '{v}' must end with .gds")
+        return v
+
+    @validator("designrulefile")
+    def check_designrulefile_exists(cls, v: Path) -> Path:
+        """Check DRC file exists"""
+        if not _check_file_exists(v):
+            raise ValidationError(f"DRC file '{v}' does not exist")
+        return v
+
+    @validator("designrulefile")
+    def check_designrulefile_filetype(cls, v: Path) -> Path:
+        """Check DRC filetype is .drc"""
+        if v.suffix != ".drc":
+            raise ValidationError(f"DRC file '{v}' must end with .drc")
+        return v
+
+    @validator("designrulefile")
+    def check_designrulefile_format(cls, v: Path) -> Path:
+        """Check DRC file is formatted correctly"""
+        _check_drcfile_format(v)
+        return v
+
+    def run_drc_on_gds(self) -> None:
+        """Runs KLayout's DRC on a gds file.
+
+        Raises
+        ------
+        FileNotFoundError
+            If GDS file or design rule file is not found.
+        RuntimeError
+            If KLayout is not found in system PATH.
+        SyntaxError
+            If the design rule file is not formatted correctly.
+        """
+        return run_drc_on_gds(self.gdsfile, self.designrulefile, self.resultsfile)
+
+    def run_drc(
+        self,
+        td_object: Union[Geometry, Structure, Simulation],
+        verbose: Optional[bool] = True,
+        **to_gds_file_kwargs,
+    ) -> None:
+        """Runs KLayout's DRC on a Tidy3D object. The object can be a :class:`.Geometry`, :class:`.Structure`, or :class:`.Simulation`.
+
+        Parameters
+        ----------
+        td_object : Union[:class:`.Geometry`, :class:`.Structure`, :class:`.Simulation`]
+            The Tidy3D object to run DRC on. Can be an object of type :class:`.Geometry`, :class:`.Structure`, or :class:`.Simulation`.
+        verbose : Optional[bool]
+            Whether to print logging. Default is True.
+        **to_gds_file_kwargs
+            Additional keyword arguments to pass to the Tidy3D object-specific ``to_gds_file()`` method.
+
+        Raises
+        ------
+        RuntimeError
+            If KLayout is not found in system PATH.
+        FileNotFoundError
+            If GDS file or design rule file is not found.
+
+        Example
+        -------
+        >>> from tidy3d.plugins.klayout.drc import DRCRunner
+        >>> from tidy3d.components.geometry import PolySlab
+        >>> vertices = [(-2, 0), (-1, 1), (0, 0.5), (1, 1), (2, 0), (0, -1)]
+        >>> geom = PolySlab(vertices=vertices, slab_bounds=(0, 0.22), axis=2)
+        >>> drc = DRCRunner(gdsfile="geom.gds", designrulefile="test.drc", resultsfile="drc_results.lyrdb")
+        >>> results = drc.run_drc(geom, z=0.1, gds_layer=0, gds_dtype=0)
+        >>> print(results)
+        """
+        return run_drc(
+            td_object,
```

Comment:
as a general rule, let's use keywords to pass these things around. eg `a=b` in the function. it's easier to see what these are passing rather than relying on order

Reply:
this pops up in a few other places.